src/main/java/org/folio/rest/impl/BLUsersAPI.java

```diff
@@ -979,10 +979,9 @@ okapiHeaders, null, handlePreviousResponse(false, false, false,
             Response permsResponse = cf.get();
             handleResponse(permsResponse, false, true, false, aRequestHasFailed, asyncResultHandler);
             if(!aRequestHasFailed[0] && permsResponse.getBody() != null){
-              //data coming in from the service isnt returned as required by the composite user schema
-              JsonObject j = new JsonObject();
-              j.put("permissions", permsResponse.getBody().getJsonArray("permissionNames"));
-              cu.setPermissions((Permissions) Response.convertToPojo(j, Permissions.class));
+              var permissionNames = permsResponse.getBody().getJsonArray("permissionNames").getList();
```

Comment:
@julianladisch  it seems like a there are alot of other calls to `convertToPojo` in this class. Also I think we identified that the memory consumption wasn't tied to this operation, but rather was due to another process outside of the scope of this module which was adding way too many perms to the user. Should this change perhaps be removed?

I think the other changes for security upgrades are still quite relevant.